Sources/Networking/Core/Upload/MultipartFormData+BodyPart.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  MultipartFormData+BodyPart.swift
+//  
+//
+//  Created by Tony Ngo on 18.06.2023.
+//
+
+import Foundation
+
+public extension MultipartFormData {
+    /// Represents an individual part of the `multipart/form-data`.
+    struct BodyPart {
+        /// The input stream containing the data of the part's body.
+        let dataStream: InputStream
+
+        /// The name parameter of the `Content-Disposition` header field.
+        let name: String
+
+        /// The size of the part's body.
+        let size: UInt64
+
+        /// An optional file parameter of the `Content-Disposition` header field. This value may be provided if the body part represents a content of a file.
+        let fileName: String?
+
+        /// An optional value of the `Content-Type` header field.
+        let mimeType: String?
+    }
+}
+
+extension MultipartFormData.BodyPart {
```

Comment:
doesn't this needs to be public as well?

Reply:
The `BodyPart` can be only created when appending data to the `MultipartFormData` and because the current implementation doesn't provide any API to work with body parts directly it wasn't needed to expose this detail to the client. 

I'm sure the implementation doesn't cover all use cases so if necessary we can certainly make it public.

Reply:
I always thought it was the other way around. I'll change it.